tests/test_libparallel.py

```diff
@@ -0,0 +1,120 @@
+import pytest
+from haddock.libs.libparallel import split_tasks, get_index_list, Worker, Scheduler
+from pathlib import Path
+from multiprocessing import Queue
+import uuid
+
+
+class Task:
+
+    def __init__(self, input):
+        self.input = input
+        self.output = None
+
+    def run(self) -> int:
+        self.output = self.input + 1
+        return self.output
+
+
+class FileTask:
+
+    def __init__(self, filename):
+        self.input_file = Path(filename)
+
+    def run(self):
+        Path(self.input_file).touch()
+
+
+@pytest.fixture
+def worker():
+    yield Worker(tasks=[Task(1), Task(2), Task(3)], results=Queue())
+
+
+@pytest.fixture
+def scheduler():
+    yield Scheduler(
+        ncores=1,
+        tasks=[
+            Task(1),
+            Task(2),
+            Task(3),
+        ],
+    )
+
+
+@pytest.fixture
+def scheduler_files():
+
```

Comment:
can you add a couple of words explaining what this test is doing?

Reply:
This is the fixture, not the test - but I'll add some lines